### PR TITLE
IGVF-1519-reverse-link

### PR DESCRIPTION
**Pull request by meenakshikagda**

### CODE REVIEW

File: src/igvfd/types/biomarker.py [hunk 1]
```diff
@@ -45,3 +53,19 @@ def unique_keys(self, properties):
         value = self.name_quantification(name=properties['name'], quantification=properties['quantification'])
         keys.setdefault('biomarker:name_quantification', []).append(value)
         return keys
+
+    @calculated_property(schema={
+        'title': 'Biomarker for',
```
Comment (ian-whaling) [inline src/igvfd/types/biomarker.py:58]: Use title case for property titles.

File: src/igvfd/types/biomarker.py [hunk 2]
```diff
@@ -45,3 +53,19 @@ def unique_keys(self, properties):
...
+        'description': 'The samples which have been confirmed to have this biomarker.',
+        'type': 'array',
+        'minItems': 1,
+        'uniqueItems': True,
+        'items': {
+            'title': 'Biomarker for',
```
Comment (ian-whaling) [inline src/igvfd/types/biomarker.py:64]: Use title case for property titles.

File: src/igvfd/types/biomarker.py [hunk 3]
```diff
@@ -45,3 +53,19 @@ def unique_keys(self, properties):
...
+            'type': ['string', 'object'],
+            'linkFrom': 'Sample.biomarkers',
```
Comment (ian-whaling) [inline src/igvfd/types/biomarker.py:66]: biomarkers is on the level of Biosample so it should link from there instead.

File: src/igvfd/types/biomarker.py [hunk 4]
```diff
@@ -45,3 +53,19 @@ def unique_keys(self, properties):
...
+        },
+        'notSubmittable': True
+    })
+    def input_to(self, request, biomarker_for):
```
Comment (ian-whaling) [inline src/igvfd/types/biomarker.py:70]: property name should be biomarker_for

File: src/igvfd/types/file.py [hunk 5]
```diff
@@ -138,6 +139,22 @@ class File(Item):
     def integrated_in(self, request, integrated_in):
         return paths_filtered_by_status(request, integrated_in)
 
+    @calculated_property(schema={
+        'title': 'Input to',
```
Comment (ian-whaling) [inline src/igvfd/types/file.py:143]: Use title case for property titles

File: src/igvfd/types/file.py [hunk 6]
```diff
@@ -138,6 +139,22 @@ class File(Item):
...
+        'description': 'The files which have been derived using this file as an input.',
+        'type': 'array',
+        'minItems': 1,
+        'uniqueItems': True,
+        'items': {
+            'title': 'Input to',
```
Comment (ian-whaling) [inline src/igvfd/types/file.py:149]: Use title case for property titles

File: src/igvfd/types/modification.py [hunk 7]
```diff
@@ -75,6 +83,22 @@ def summary(self, request, cas, cas_species, modality, fused_domain=None, tagged
 
         return summary
 
+    @calculated_property(schema={
+        'title': 'Samples modified',
```
Comment (ian-whaling) [inline src/igvfd/types/modification.py:87]: Use title case for property titles

File: src/igvfd/types/modification.py [hunk 8]
```diff
@@ -75,6 +83,22 @@ def summary(self, request, cas, cas_species, modality, fused_domain=None, tagged
...
+        'description': 'The samples which have been modified with this modification.',
+        'type': 'array',
+        'minItems': 1,
+        'uniqueItems': True,
+        'items': {
+            'title': 'Samples modified',
```
Comment (ian-whaling) [inline src/igvfd/types/modification.py:93]: Use title case for property titles

File: src/igvfd/types/modification.py [hunk 9]
```diff
@@ -75,6 +83,22 @@ def summary(self, request, cas, cas_species, modality, fused_domain=None, tagged
...
+            'type': ['string', 'object'],
+            'linkFrom': 'Sample.modifications',
```
Comment (ian-whaling) [inline src/igvfd/types/modification.py:95]: modifications is on the level of Biosample so it should link from there instead.

File: src/igvfd/types/treatment.py [hunk 10]
```diff
@@ -48,3 +56,19 @@ def summary(self, treatment_term_name, amount=None, amount_units=None, duration=
...
+    @calculated_property(schema={
+        'title': 'Samples Treated',
+        'description': 'The samples which have been treated using this treatment.',
+        'type': 'array',
+        'minItems': 1,
+        'uniqueItems': True,
+        'items': {
+            'title': 'Samples Treated',
+            'type': ['string', 'object'],
+            'linkFrom': 'Sample.treatments',
```
Comment (ian-whaling) [inline src/igvfd/types/treatment.py:69]: treatments is on the level of Biosample so it should link from there instead.

File: src/igvfd/types/treatment.py [hunk 11]
```diff
@@ -48,3 +56,19 @@ def summary(self, treatment_term_name, amount=None, amount_units=None, duration=
...
+        },
+        'notSubmittable': True
+    })
+    def input_to(self, request, samples_treated):
```
Comment (ian-whaling) [inline src/igvfd/types/treatment.py:73]: property name should be samples_treated

File: src/igvfd/types/treatment.py [hunk 12]
```diff
@@ -25,6 +30,9 @@ class Treatment(Item):
         Path('sources', include=['@id', 'title']),
         Path('submitted_by', include=['@id', 'title']),
     ]
+    rev = {
+        'samples_treated': ('Sample', 'treatments')
```
Comment (ian-whaling) [inline src/igvfd/types/treatment.py:34]: treatments is on the level of Biosample so it should link from there instead.

File: src/igvfd/types/modification.py [hunk 13]
```diff
@@ -24,6 +29,9 @@ class Modification(Item):
         Path('lab', include=['@id', 'title']),
         Path('submitted_by', include=['@id', 'title']),
     ]
+    rev = {
+        'samples_modified': ('Sample', 'modifications')
```
Comment (ian-whaling) [inline src/igvfd/types/modification.py:33]: modifications is on the level of Biosample so it should link from there instead.

File: src/igvfd/types/biomarker.py [hunk 14]
```diff
@@ -24,6 +29,9 @@ class Biomarker(Item):
         Path('lab', include=['@id', 'title']),
         Path('submitted_by', include=['@id', 'title']),
     ]
+    rev = {
+        'biomarker_for': ('Sample', 'biomarkers')
```
Comment (ian-whaling) [inline src/igvfd/types/biomarker.py:33]: biomarkers is on the level of Biosample so it should link from there instead.

**Review by ian-whaling — CHANGES_REQUESTED** [review]
Some changes needed to function names, title case, and level of reverse link

**Comment by meenakshikagda** [comment]
Created a new PR https://github.com/IGVF-DACC/igvfd/pull/969

**Pull request closed without merging**